gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,229 @@
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/gpii/universal/LICENSE.txt
+ */
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.defaults("gpii.flowManager.capture", {
+    gradeNames: ["fluid.component"],
+    events: {
+        // Pseudoevents for transforming promise chains to fetch the solutions on the current device
+        // and capture the settings for the device.
+        onSolutionsForCurrentDevice: null,
+        onCaptureSettingsForCurrentDevice: null
+    },
+    listeners: {
+        // Begin declaration of Promise Chain for onSolutionsForCurrentDevice
+        "onSolutionsForCurrentDevice.getDeviceContextPromise": {
+            funcName: "gpii.lifecycleManager.getDeviceContextPromise",
+            args: ["{flowManager}.deviceReporter"]
+        },
+        "onSolutionsForCurrentDevice.getSolutionsPromise": {
+            funcName: "gpii.flowManager.getSolutionsPromise",
+            args: [ "{flowManager}.solutionsRegistryDataSource", "{arguments}.0"],
+            priority: "after:getDeviceContextPromise"
+        },
+        "onSolutionsForCurrentDevice.solutionsRegistryEntriesToPromise": {
+            funcName: "fluid.toPromise",
+            args: ["{arguments}.0.solutionsRegistryEntries"],
+            priority: "after:getSolutionsPromise"
+        },
+        // Begin declaration of Promise Chain for onCaptureSettingsForCurrentDevice
+        "onCaptureSettingsForCurrentDevice.getInstalledSolutionsForCurrentDevice": {
+            func: "{that}.getInstalledSolutionsForCurrentDevice"
+        },
+        "onCaptureSettingsForCurrentDevice.captureSystemSettings": {
+            func: "{that}.captureSystemSettings",
+            args: ["{arguments}.0", "{arguments}.1"],
+            priority: "after:getInstalledSolutionsForCurrentDevice"
+        },
+        "onCaptureSettingsForCurrentDevice.formatRawCapturedSettings": {
+            func: "gpii.flowManager.formatRawCapturedSettings",
+            args: ["{arguments}.0"],
+            priority: "after:captureSystemSettings"
+        }
+    },
+    invokers: {
+        getInstalledSolutionsForCurrentDevice: {
+            funcName: "gpii.flowManager.getInstalledSolutionsForCurrentDevice",
+            args: ["{that}"]
+        },
+        getSystemSettingsCapture: {
+            funcName: "gpii.flowManager.getSystemSettingsCapture",
+            args: ["{that}.events.onCaptureSettingsForCurrentDevice", "{arguments}.0"] // options
+        },
+        captureSystemSettings: {
+            funcName: "gpii.flowManager.captureSystemSettings",
+            args: ["{lifecycleManager}.read", "{arguments}.0", "{arguments}.1"] // solutionsRegistryEntries, options
+        }
+    }
+});
+
+gpii.flowManager.getInstalledSolutionsForCurrentDevice = function (that) {
```

Comment:
This function can be directly declared in the "invokers" block, to replace line 58-59.

The same comment applies to the next function `gpii.flowManager.getSystemSettingsCapture`.

Reply:
Moved both of these into just invokers, and left a jsdoc for the main one.